docs: remove internal links in common directory (#1162)
canonical/craft-parts: docs/common/craft-parts/reference/plugins/nil_plugin.rst

How it works
------------

> This plugin does nothing. It serves as a *noop* when there is a need to only use default
> part properties.

Reviewer: ```suggestion
This plugin does nothing. It serves as a *noop* when only the default part properties
are needed.
```
Out of scope for this PR
